src/App.tsx

```diff
@@ -15,36 +16,90 @@
   'Garlic',
 ];
 
+const SORT_FILD_ALPHABETICCALY = 'alphabeticcaly';
+const SORT_FILD_LENGTH = 'length';
+
+function getPreparedGoods(goods: string[], sortField: string, rever: boolean) {
+  let newVisibleGoods = [...goods];
+
+  if (sortField === SORT_FILD_ALPHABETICCALY) {
+    newVisibleGoods.sort();
+  }
+
+  if (sortField === SORT_FILD_LENGTH) {
+    newVisibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (rever) {
+    return newVisibleGoods.reverse();
+  }
+
+  if (sortField === '') {
+    newVisibleGoods = [...goods];
+  }
+
+  return newVisibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [rever, setRever] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, rever);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SORT_FILD_ALPHABETICCALY,
+          })}
+          onClick={() => setSortField(SORT_FILD_ALPHABETICCALY)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SORT_FILD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FILD_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': rever === false,
+          })}
+          onClick={() => setRever(!rever)}
```

Comment:
The function `setRever` is used here, but it seems to be a typo. Consider renaming it to `setReverse` for clarity.